src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/AutoConfirmUser/AutomaticallyConfirmOrganizationUsersValidator.cs

```diff
@@ -0,0 +1,120 @@
+    ITwoFactorIsEnabledQuery twoFactorIsEnabledQuery,
+    IPolicyService policyService,
+    IPolicyRequirementQuery policyRequirementQuery
+    ) : IAutomaticallyConfirmOrganizationUsersValidator
+{
+    public async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>> ValidateAsync(
+        AutomaticallyConfirmOrganizationUserRequestData request)
+    {
+        var orgUser = request.OrganizationUser;
+
+        if (orgUser.Status != OrganizationUserStatusType.Accepted)
+        {
+            return Invalid(request, new UserIsNotAccepted());
+        }
+
+        if (orgUser.OrganizationId != request.Organization.Id)
+        {
+            return Invalid(request, new OrganizationUserIdIsInvalid());
+        }
+
+        return await OrganizationUserOwnershipValidationAsync(request)
+            .ThenAsync(OrganizationUserConformsToOrganizationPoliciesAsync);
+    }
+
+    private async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>> OrganizationUserOwnershipValidationAsync(
+        AutomaticallyConfirmOrganizationUserRequestData request) =>
+        request.Organization.PlanType is not PlanType.Free
+        || await organizationUserRepository.GetCountByFreeOrganizationAdminUserAsync(request.UserId) == 0
+            ? Valid(request)
+            : Invalid(request, new UserToConfirmIsAnAdminOrOwnerOfAnotherFreeOrganization());
+
+    private async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>>
+        OrganizationUserConformsToOrganizationPoliciesAsync(AutomaticallyConfirmOrganizationUserRequestData request)
+    {
+        return await OrganizationUserConformsToTwoFactorRequiredPolicyAsync(request)
+            .ThenAsync(OrganizationUserConformsToSingleOrgPolicyAsync);
+    }
+
+    private async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>> OrganizationUserConformsToTwoFactorRequiredPolicyAsync(
+            AutomaticallyConfirmOrganizationUserRequestData request)
+    {
+        if ((await twoFactorIsEnabledQuery.TwoFactorIsEnabledAsync([request.UserId]))
+            .Any(x => x.userId == request.UserId && x.twoFactorIsEnabled))
+        {
+            return Valid(request);
+        }
+
+        if (!featureService.IsEnabled(FeatureFlagKeys.PolicyRequirements))
+        {
+            return (await policyService.GetPoliciesApplicableToUserAsync(request.UserId,
+                    PolicyType.TwoFactorAuthentication))
+                .Any(p => p.OrganizationId == request.Organization.Id)
+                    ? Invalid(request, new UserDoesNotHaveTwoFactorEnabled())
+                    : Valid(request);
+        }
+
+        return (await policyRequirementQuery.GetAsync<RequireTwoFactorPolicyRequirement>(request.UserId))
+            .IsTwoFactorRequiredForOrganization(request.Organization.Id)
+                ? Invalid(request, new UserDoesNotHaveTwoFactorEnabled())
+                : Valid(request);
+    }
```

Comment:
Oh man this nesting of feature flags is rough, but also my fault. `PolicyRequirement`s is going to change substantially, so my guidance for now is to use it for new logic where it's needed (like `OrganizationDataOwnership` - it has all the new "is a default collection required" logic that doesn't exist anywhere else) - but not for old code that is still using the legacy `policyService` in prod today (like 2FA).